components/ListView/SourcesListItem.js

```diff
@@ -0,0 +1,131 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { FormattedMessage, defineMessages, injectIntl, intlShape } from "react-intl";
+
+const messages = defineMessages({
+  edit: {
+    defaultMessage: "Edit Source"
+  },
+  kebab: {
+    defaultMessage: "Source Actions"
+  },
+  remove: {
+    defaultMessage: "Remove Source"
+  },
+  baseurl: {
+    defaultMessage: "yum repository"
+  },
+  mirrorlist: {
+    defaultMessage: "mirrorlist"
+  },
+  metalink: {
+    defaultMessage: "metalink"
+  }
+});
+
+class SourcesListItem extends React.Component {
+  constructor(props) {
+    super(props);
+    this.editButton = React.createRef();
+  }
+  componentDidMount() {
+    if (this.props.edited === this.props.source.name) {
```

Comment:
This 'edited' property is just showing if a source is editable? If so this can be removed and this can be checked from the edit property. If it's not undefined, means there is a handler for the edit source, so it can be edited.

Reply:
This is used for setting focus on the Edit button for the list item that was just edited. Since the edit button displays the form, the same button should receive focus when the list of sources displays again.